Creating quick start guide as part of overall docs improvements (#249)
NCAR/geocat-comp: docs/getting_started/overview.rst

.. currentmodule:: geocat.comp

.. _overview:

Overview
========
GeoCAT-comp is the computational component of the `GeoCAT project <https://geocat.ucar.edu/>`_. It houses Python
implementations of NCL's non-WRF computational routines and additional geoscientific analysis functions that go beyond

> what NCL offers. It is a principle component of NCAR's `Pivot to Python <https://www.ncl.ucar.edu/Document/Pivot_to_Python/>`_.

Reviewer: "... NCAR's Pivot to Python initiative."

Why GeoCAT-comp?

> ----------------

Reviewer: There are much more important motivations to start off this "Why GeoCAT-comp?" section than the dimension requirements that is discussed in this first paragraph. 

I think the actual first paragraph should be Python-related benefits. For this, some of those significant ones (i.e. easy-to-learn, platform-independent, huge package ecosystem, etc.) can be mentioned, then the readers could be directed to Pivot-to-Python link again for details.

"Scalability" needs to be the next reason. Something like the following could be great as a second paragraph:

"GeoCAT-comp achieves scalability through parallel computation that was never possible with NCL, enabling geoscientists to tackle analysis workflows that were not previously possible due to the volumes of data involved."

After that, we can switch the order of the current first and second paragraphs. This is because we are using Xarray essentially for compatibility with the huge Pangeo ecosystem for data I/O, recognition, processing, etc. The dimension flexibility of Xarray is an added convenience here that needs to be discussed after our essential motivation with using Xarray.

Reviewer: Maybe after the first paragraph, we can even mention our "open development" philiosophy here as a separate paragraph? Saying something like: " GeoCAT-comp is a geoscience data analysis-focused Python package that is dedicated to address and community needs in an open development mindset." Then we can describe what our open development approach is and how it is realized. 

I'd highly recommend to use the GeoCAT's SciPy 2022 paper for populating this section.

be in a particular order. This resulted in functions with rigid data format requirements or multiple versions of the
same function for different dimension orders.

To avoid those data format requirements, GeoCAT-comp uses `xarray <https://docs.xarray.dev/en/stable/getting-started-guide/why-xarray.html#what-labels-enable>`_
data structures so users can refer to dimensions by name rather than index. This removes the need for different function
versions which makes the finding the right function easier for the user. This also adds flexibility so that the user
need not spend as much time time wrangling their data into the correct format.

Beyond recreating NCL routines in Python, GeoCAT-comp's open source nature allows users to make requests for or contribute
new functions that don't exist in NCL. As advances are made in the realm of geoscience, new tools will be needed to
analyze new datasets. GeoCAT-comp is the home for those tools!

GeoCAT-f2py
-----------
While our goal is to recreate NCL functions in pure Python, translating some NCL routines is challenging and time

> consuming. To ensure GeoCAT users have access to those functions while we work on full Python versions, the Fortran

Reviewer: either "the Fortran language" or "Fortran"

Reviewer 2: I think this makes sense in the context of the phrase "the Fortran they are based upon"

Author: I can change it to "the Fortran code they are based upon" if you think that makes it more clear @erogluorhan

Reviewer: Yes!

> they are based upon is wrapped in Python in the GeoCAT-f2py (Fortran 2 Python) package. GeoCAT-f2py is accessible

Reviewer 2: I think we should mention that we're using the f2py package here

> through the GeoCAT-comp namespace, so there is no need for users to install GeoCAT-f2py directly. Information about

Reviewer 2: Also, do you think this terminology is too complicated for the quickstart guide?

Author: I tried to clarify it a bit in my most recent commit. Let me know if it needs improvement

> GeoCAT-f2py can be found on the `package's homepage <https://geocat-f2py.readthedocs.io/en/latest/#>`_.

Reviewer 2: ```suggestion
GeoCAT-f2py can be found on the `package's homepage <https://geocat-f2py.readthedocs.io/en/>`_.
```

I think we should let our RTD settings decide to route to latest or stable

There are syntactical benefits to using the Pangeo stack, once of which is indexing by label. NCL requires the
dimensions of input data to be in a particular order. This results in functions with rigid data format requirements and
multiple versions of the same function for different dimension orders. By using `xarray <https://docs.xarray.dev/>`_,
GeoCAT-comp avoids those data format requirements. The xarray data structures allow users to
`refer to dimensions by label <https://docs.xarray.dev/en/stable/getting-started-guide/why-xarray.html#what-labels-enable>`_
rather than index, thereby removing the need for multiple versions of the same function. This makes it easier for the
user to find what they need and adds flexibility so that the user need not spend as much time wrangling their data
into the correct format.

GeoCAT-comp's open-source nature allows for more community engagement than a traditional software development workflow
does. As advances are made in the realm of geoscience, new tools will be needed to analyze new datasets. We are dedicated
to addressing user needs and encourage users to submit feature requests and contributions to our GitHub as well as

> participate in discussions. See our `support <support>`_ page for info on how to submit bug reports and requests and how to get involved.

Reviewer 2: ```suggestion
participate in discussions. See our `support <support>`_ page for info on how to submit bug report, feature requests, and how to get involved.
```

Reviewer 2: Yeah, that sounds better!

> participate in discussions. See our `support <support>`_ page for info on how to submit bug reports and requests and how to get involved.

Reviewer 2: This link here actually doesn't work. I think you might want to do something like :ref:`support`

GeoCAT-f2py
-----------
While our goal is to recreate NCL functions in pure Python, translating some NCL routines is challenging and time
consuming. To ensure GeoCAT users have access to those functions while we work on full Python versions, the Fortran code
they are based upon is wrapped in Python in the GeoCAT-f2py (Fortran 2 Python) package. GeoCAT-f2py is imported by
GeoCAT-comp and included in the GeoCAT-comp namespace. Simply, this means that GeoCAT-f2py functions can be called
through GeoCAT-comp, so there is no need for users to import GeoCAT-f2py directly in their scripts. Information about

> GeoCAT-f2py can be found on the `package's homepage <https://geocat-f2py.readthedocs.io/en/>`_.

Reviewer 2: Ahh, my bad, apparently RTD doesn't like that link. Let's do:
```suggestion
GeoCAT-f2py can be found on the `package's homepage <https://geocat-f2py.readthedocs.io/>`_.
```

`NCL <https://www.ncl.ucar.edu/>`_, the prior language of choice for geoscience work, has been put into maintenance mode
in favor of Python. This is due to Python's easy-to-learn, open-source nature and the benefits that provides. There are
a plethora of scientific analysis packages designed for general use or for niche applications. Numerous tutorials exist
for learning Python basics and for data analysis workflows. Python also enables scalability through parallel computation
which was never possible with NCL, enabling geoscientists to tackle analysis workflows on large volumes of data.

GeoCAT-comp draws from well-established analysis packages like `NumPy <https://numpy.org/>`_, `xarray <https://docs.xarray.dev/>`_,
`SciPy <https://scipy.org/>`_, and `MetPy <https://unidata.github.io/MetPy/>`_ to recreate and expand upon NCL
functions in pure Python. With so many tools being used under the hood, GeoCAT-comp users have access to geoscience
specific computational tools without needing extensive working knowledge of `Pangeo <https://pangeo.io/>`_ stack.

> There are syntactical benefits to using the Pangeo stack, once of which is indexing by label. NCL requires the

Reviewer: typo: "one of which"
